sql/session.go

```diff
@@ -0,0 +1,22 @@
+package sql
+
+import (
+	"context"
+)
+
+// Session holds context and session data
+type Session interface {
+	context.Context
+}
+
+// BaseSession is the basic session type
+type BaseSession struct {
+	context.Context
+}
+
+// NewBaseSession creates a new basic session
+func NewBaseSession() Session {
```

Comment:
Maybe this should accept a context. When we have custom implementations of the session they might want to do a `NewBaseSession` but have the ability to pass in their own context with cancellations and so on.

Reply:
Changed to accept a context.